src/Mine.cpp

```diff
@@ -346,7 +346,7 @@ void Mine::deserialize(NAS2D::Xml::XmlElement* _ti)
 	mVeins.resize(depth);
 	for (XmlNode* vein = _ti->firstChild(); vein != nullptr; vein = vein->nextSibling())
 	{
-		MineVein _mv = MineVein{0, 0, 0, 0};
+		auto _mv = MineVein{0, 0, 0, 0};
```

Comment:
Was thinking that since this is a declaration, it could use initialization:
```cpp
MineVein vein{0, 0, 0, 0};
```

That might be a better general solution than default initialization, followed by assignment to overwrite the default values. Not that it matters since `MineVein` is just a `typedef` for `std::array<int, 4>`, so default initialization should just do nothing and leave it uninitialized, which is then overwritten by the assignment.

Same thought for both updated lines.

Reply:
The language specifies that declaration of the form `Foo a = Foo()` or `Foo a = Foo{}` is not an assignment. It is initialization. The copy constructor is never called.

See: https://en.cppreference.com/w/cpp/language/initialization

Reply:
Right. So style choice then.

Reply:
Yes. "Almost always auto". :)